internal/oc/release.go

```diff
@@ -1,64 +1,55 @@
 package oc
 
 import (
-	"bytes"
 	"errors"
 	"fmt"
 	"io/ioutil"
 	"os"
-	"os/exec"
 	"path/filepath"
 	"strings"
 
+	"github.com/hashicorp/go-version"
+	"github.com/openshift/assisted-service/internal/common"
 	"github.com/sirupsen/logrus"
 )
 
 //go:generate mockgen -source=release.go -package=oc -destination=mock_release.go
 type Release interface {
 	GetMCOImage(log logrus.FieldLogger, releaseImage string, releaseImageMirror string, pullSecret string) (string, error)
+	GetOpenshiftVersion(log logrus.FieldLogger, releaseImage string, releaseImageMirror string, pullSecret string) (string, error)
+	GetMajorMinorVersion(log logrus.FieldLogger, releaseImage string, releaseImageMirror string, pullSecret string) (string, error)
 	Extract(log logrus.FieldLogger, releaseImage string, releaseImageMirror string, cacheDir string, pullSecret string) (string, error)
 }
 
-type release struct {
+type Executer interface {
+	Execute(command string, args ...string) (stdout string, stderr string, exitCode int)
+	TempFile(dir, pattern string) (f *os.File, err error)
 }
 
-func NewRelease() Release {
-	return &release{}
-}
+type ReleaseExecuter struct{}
```

Comment:
`ReleaseExecuter` and `Executer` don't seem like they belong in this package. Can you move them to `common`?
Then we also don't need the weird call proxying in `ReleaseExecuter.Execute()`

Reply:
Obviously rename `ReleaseExecuter`, but it would do exactly the same thing.